R/chartutils.R

```diff
@@ -1222,22 +1225,27 @@ setTicks <- function(minimum, maximum, distance, reversed = FALSE,
     if (!is.null(data))
     {
         is.bar <- grepl("Bar", type) && !grepl("Stacked", type)
-        if (is.null(minimum))
-            minimum <- min(0, min(data, na.rm = TRUE))
-        if (is.null(maximum))
-            maximum <- max(data, na.rm = TRUE)
+        dat_min <- min(0, data, na.rm = TRUE)
+        dat_max <- max(data, na.rm = TRUE)
+        if (dat_min == dat_max)
+            dat_max <- 0
```

Comment:
Wouldn't this cause `dat_max` to be less than `dat_min`, if they are both the same positive value? The `pad` would then become negative.

Reply:
Ah of course